File: Arcade/Intro/Java/IsIPv4Address.java
final class IsIPv4Address {

	boolean isIPv4Address(String inputString) {
	    String[] split = inputString.split("\\.");
	    if(split.length != 4) return false;
	    for(String s : split) {
	        if(s.matches("\\d*[a-z]+\\d*") || s.length() == 0) return false;
	        if(s.length() > 1 && (s.charAt(0) == '0' || new Long(s) > 255)) {
	            return false;   
	        }
	    }
	    return true;
	}
}

File: Arcade/Intro/javascript/isIPv4Address.js
function isIPv4Address(inputString) {
    let split = inputString.split(".");
    
    if(split.length != 4) return false;
    
    for(let o of split) {
        let e = o.match("\d*[a-z]+\d*");
        if(e != null) if(e[0].length > 0) return false;
        
        if(o.length == 0) return false;
        
        if(o.length > 1 && (o.charAt(0) == '0' || parseInt(o) > 255))
            return false;
    }
    return true;
}